diff --git a/CMakeLists.txt b/CMakeLists.txt
index b3d968963..31cfae3af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -383,8 +383,8 @@ endif()
 # boost::process was introduced first in version 1.64.0,
 # boost::beast::detail::base64 was introduced first in version 1.66.0
 set(MINIMUM_BOOST_VERSION "1.83.0")
-set(_boost_components "system;filesystem;thread;log;locale;regex;chrono;atomic;date_time;iostreams")
-find_package(Boost ${MINIMUM_BOOST_VERSION} REQUIRED COMPONENTS ${_boost_components})
+set(_boost_components "filesystem;thread;log;locale;regex;chrono;atomic;date_time;iostreams")
+find_package(Boost ${MINIMUM_BOOST_VERSION} REQUIRED CONFIG COMPONENTS ${_boost_components})
 
 add_library(boost_libs INTERFACE)
 add_library(boost_headeronly INTERFACE)
diff --git a/cmake/modules/FindOpenVDB.cmake b/cmake/modules/FindOpenVDB.cmake
index 2a61b21a2..46d425eda 100644
--- a/cmake/modules/FindOpenVDB.cmake
+++ b/cmake/modules/FindOpenVDB.cmake
@@ -123,7 +123,7 @@ if (OPENVDB_FIND_MODULE_PATH)
 endif ()
 # ###########################################################################
 
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.5)
 # Monitoring <PackageName>_ROOT variables
 if(POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
diff --git a/deps/Assimp/Assimp.cmake b/deps/Assimp/Assimp.cmake
index 1c38b3399..f4ce6fd39 100644
--- a/deps/Assimp/Assimp.cmake
+++ b/deps/Assimp/Assimp.cmake
@@ -2,6 +2,7 @@ bambustudio_add_cmake_project(Assimp
     URL "https://github.com/assimp/assimp/archive/refs/tags/v5.4.3.tar.gz"
     URL_HASH SHA256=66dfbaee288f2bc43172440a55d0235dfc7bf885dda6435c038e8000e79582cb
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DASSIMP_BUILD_TESTS=OFF
         -DASSIMP_BUILD_SAMPLES=OFF
         -DASSIMP_BUILD_ASSIMP_TOOLS=OFF
diff --git a/deps/Blosc/Blosc.cmake b/deps/Blosc/Blosc.cmake
index 93e4564e8..ca752d036 100644
--- a/deps/Blosc/Blosc.cmake
+++ b/deps/Blosc/Blosc.cmake
@@ -23,6 +23,7 @@ bambustudio_add_cmake_project(Blosc
     # Patching upstream does not work this way with git version 2.28 installed on mac worker
     # PATCH_COMMAND  ${GIT_EXECUTABLE} apply --ignore-space-change --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/blosc-mods.patch
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DCMAKE_POSITION_INDEPENDENT_CODE=ON
         -DBUILD_SHARED=${_build_shared} 
         -DBUILD_STATIC=${_build_static}
diff --git a/deps/Boost/Boost.cmake b/deps/Boost/Boost.cmake
index bc66a6b74..4f2cda173 100644
--- a/deps/Boost/Boost.cmake
+++ b/deps/Boost/Boost.cmake
@@ -38,6 +38,7 @@ bambustudio_add_cmake_project(Boost
     LIST_SEPARATOR |
     PATCH_COMMAND git apply --verbose --ignore-space-change --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/0001-FIX-OBS-cannot-start-streaming-on-MAC.patch
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DBOOST_EXCLUDE_LIBRARIES:STRING=contract|fiber|numpy|wave|test
         -DBOOST_LOCALE_ENABLE_ICU:BOOL=OFF # do not link to libicu, breaks compatibility between distros
         -DBUILD_TESTING:BOOL=OFF
diff --git a/deps/CMakeLists.txt b/deps/CMakeLists.txt
index 179973f8d..4516819c6 100644
--- a/deps/CMakeLists.txt
+++ b/deps/CMakeLists.txt
@@ -21,7 +21,7 @@
 #
 
 project(BambuStudio-deps)
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.5)
 
 include(ExternalProject)
 include(ProcessorCount)
diff --git a/deps/CURL/CURL.cmake b/deps/CURL/CURL.cmake
index 6de704ff8..7c11d68c1 100644
--- a/deps/CURL/CURL.cmake
+++ b/deps/CURL/CURL.cmake
@@ -65,6 +65,7 @@ bambustudio_add_cmake_project(CURL
   # PATCH_COMMAND       ${GIT_EXECUTABLE} checkout -f -- . && git clean -df && 
   #                     ${GIT_EXECUTABLE} apply --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/curl-mods.patch
   CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
     -DBUILD_TESTING:BOOL=OFF
     -DBUILD_CURL_EXE:BOOL=OFF
     -DCMAKE_POSITION_INDEPENDENT_CODE=ON
diff --git a/deps/Cereal/Cereal.cmake b/deps/Cereal/Cereal.cmake
index 1bd9e3f82..aa757ef31 100644
--- a/deps/Cereal/Cereal.cmake
+++ b/deps/Cereal/Cereal.cmake
@@ -2,6 +2,7 @@ bambustudio_add_cmake_project(Cereal
     URL "https://github.com/USCiLab/cereal/archive/refs/tags/v1.3.0.zip"
     URL_HASH SHA256=71642cb54658e98c8f07a0f0d08bf9766f1c3771496936f6014169d3726d9657
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DJUST_INSTALL_CEREAL=ON
         -DSKIP_PERFORMANCE_COMPARISON=ON
         -DBUILD_TESTS=OFF
diff --git a/deps/EXPAT/expat/CMakeLists.txt b/deps/EXPAT/expat/CMakeLists.txt
index 585a678b8..30ee5234b 100644
--- a/deps/EXPAT/expat/CMakeLists.txt
+++ b/deps/EXPAT/expat/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 
 project(EXPAT)
 
diff --git a/deps/FREETYPE/FREETYPE.cmake b/deps/FREETYPE/FREETYPE.cmake
index d4b5e0811..92c5ec554 100644
--- a/deps/FREETYPE/FREETYPE.cmake
+++ b/deps/FREETYPE/FREETYPE.cmake
@@ -16,6 +16,7 @@ bambustudio_add_cmake_project(FREETYPE
     #DEPENDS ${ZLIB_PKG}
     #"${_patch_step}"
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
 	-D BUILD_SHARED_LIBS=${library_build_shared}
 	${_ft_disable_zlib}
         -D FT_DISABLE_BZIP2=TRUE
diff --git a/deps/GLEW/glew/CMakeLists.txt b/deps/GLEW/glew/CMakeLists.txt
index 513ee4ebb..e298d9750 100644
--- a/deps/GLEW/glew/CMakeLists.txt
+++ b/deps/GLEW/glew/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 project(GLEW)
 
 find_package(OpenGL REQUIRED)
diff --git a/deps/GLFW/GLFW.cmake b/deps/GLFW/GLFW.cmake
index 01f1d661d..c6d35b83c 100644
--- a/deps/GLFW/GLFW.cmake
+++ b/deps/GLFW/GLFW.cmake
@@ -17,6 +17,7 @@ bambustudio_add_cmake_project(GLFW
     URL_HASH SHA256=e02d956935e5b9fb4abf90e2c2e07c9a0526d7eacae8ee5353484c69a2a76cd0
     #DEPENDS dep_Boost
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DBUILD_SHARED_LIBS=${_build_shared} 
         -DGLFW_BUILD_DOCS=OFF
         -DGLFW_BUILD_EXAMPLES=OFF
diff --git a/deps/JPEG/JPEG.cmake b/deps/JPEG/JPEG.cmake
index b95345146..640b9ed13 100644
--- a/deps/JPEG/JPEG.cmake
+++ b/deps/JPEG/JPEG.cmake
@@ -22,6 +22,7 @@ bambustudio_add_cmake_project(JPEG
     URL_HASH SHA256=d6d99e693366bc03897677650e8b2dfa76b5d6c54e2c9e70c03f0af821b0a52f
     DEPENDS ${ZLIB_PKG}
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DENABLE_SHARED=OFF
         -DENABLE_STATIC=ON
         ${jpeg_flag}
diff --git a/deps/NLopt/NLopt.cmake b/deps/NLopt/NLopt.cmake
index daa7ac7fa..3d5f4bd1c 100644
--- a/deps/NLopt/NLopt.cmake
+++ b/deps/NLopt/NLopt.cmake
@@ -2,6 +2,7 @@ bambustudio_add_cmake_project(NLopt
   URL "https://github.com/stevengj/nlopt/archive/v2.5.0.tar.gz"
   URL_HASH SHA256=c6dd7a5701fff8ad5ebb45a3dc8e757e61d52658de3918e38bab233e7fd3b4ae
   CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
     -DNLOPT_PYTHON:BOOL=OFF
     -DNLOPT_OCTAVE:BOOL=OFF
     -DNLOPT_MATLAB:BOOL=OFF
diff --git a/deps/OCCT/OCCT.cmake b/deps/OCCT/OCCT.cmake
index 7afaa8bf2..0c146a652 100644
--- a/deps/OCCT/OCCT.cmake
+++ b/deps/OCCT/OCCT.cmake
@@ -15,6 +15,7 @@ bambustudio_add_cmake_project(OCCT
     #DEPENDS dep_Boost
     #DEPENDS dep_FREETYPE
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DBUILD_LIBRARY_TYPE=${library_build_type}
         -DUSE_TK=OFF
         -DUSE_TBB=OFF
diff --git a/deps/OpenCSG/CMakeLists.txt.in b/deps/OpenCSG/CMakeLists.txt.in
index 37da0f789..286f2a6cf 100644
--- a/deps/OpenCSG/CMakeLists.txt.in
+++ b/deps/OpenCSG/CMakeLists.txt.in
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 
 project(OpenCSG)
 
diff --git a/deps/OpenCV/OpenCV.cmake b/deps/OpenCV/OpenCV.cmake
index 8ee91434a..c6e9d23a9 100644
--- a/deps/OpenCV/OpenCV.cmake
+++ b/deps/OpenCV/OpenCV.cmake
@@ -50,6 +50,7 @@ bambustudio_add_cmake_project(OpenCV
     URL_HASH SHA256=1ec1cba65f9f20fe5a41fda1586e01c70ea0c9a6d7b67c9e13edf0cfe2239277
     PATCH_COMMAND git apply ${OpenCV_DIRECTORY_FLAG} --verbose --ignore-space-change --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/0001-OpenCV-fix.patch ${CMAKE_CURRENT_LIST_DIR}/0002-clang19-macos.patch
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
     -DBUILD_SHARED_LIBS=0
        -DBUILD_PERE_TESTS=OFF
        -DBUILD_TESTS=OFF
diff --git a/deps/OpenEXR/OpenEXR.cmake b/deps/OpenEXR/OpenEXR.cmake
index d2e8dd371..3a13c2054 100644
--- a/deps/OpenEXR/OpenEXR.cmake
+++ b/deps/OpenEXR/OpenEXR.cmake
@@ -15,6 +15,7 @@ if (APPLE AND IS_CROSS_COMPILE)
         DOWNLOAD_DIR        ${DEP_DOWNLOAD_DIR}/OpenEXR
         ${_openxr_list_sep}
         CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
             -DCMAKE_INSTALL_PREFIX:STRING=${DESTDIR}/usr/local
             -DBUILD_SHARED_LIBS:BOOL=OFF
             -DCMAKE_POSITION_INDEPENDENT_CODE=ON
@@ -43,6 +44,7 @@ bambustudio_add_cmake_project(OpenEXR
     DEPENDS ${ZLIB_PKG}
     GIT_TAG v2.5.5
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DCMAKE_POSITION_INDEPENDENT_CODE=ON
         -DBUILD_TESTING=OFF
         -DPYILMBASE_ENABLE:BOOL=OFF
diff --git a/deps/OpenVDB/OpenVDB.cmake b/deps/OpenVDB/OpenVDB.cmake
index e6816a98a..7069e7d5c 100644
--- a/deps/OpenVDB/OpenVDB.cmake
+++ b/deps/OpenVDB/OpenVDB.cmake
@@ -24,6 +24,7 @@ bambustudio_add_cmake_project(OpenVDB
     # URL_HASH SHA256=48C2CFA9853B58FA86282DF1F83F0E99D07858CC03EB2BA8227DC447A830100A
     DEPENDS dep_TBB ${BLOSC_PKG} dep_OpenEXR ${BOOST_PKG}
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DCMAKE_POSITION_INDEPENDENT_CODE=ON 
         -DOPENVDB_BUILD_PYTHON_MODULE=OFF
         -DUSE_BLOSC=ON
diff --git a/deps/PNG/PNG.cmake b/deps/PNG/PNG.cmake
index 11ffd832e..51ea87ac0 100644
--- a/deps/PNG/PNG.cmake
+++ b/deps/PNG/PNG.cmake
@@ -14,6 +14,7 @@ if(APPLE AND IS_CROSS_COMPILE)
         PATCH_COMMAND       ${GIT_EXECUTABLE} checkout -f -- . && git clean -df &&
                             ${GIT_EXECUTABLE} apply --whitespace=fix ${CMAKE_CURRENT_LIST_DIR}/macos-arm64.patch ${CMAKE_CURRENT_LIST_DIR}/0002-clang19-macos.patch
         CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
             -DPNG_SHARED=OFF
             -DPNG_STATIC=ON
             -DPNG_PREFIX=prusaslicer_
@@ -34,6 +35,7 @@ set(_patch_step "")
         DEPENDS ${ZLIB_PKG}
         "${_patch_step}"
         CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
             -DPNG_SHARED=OFF
             -DPNG_STATIC=ON
             -DPNG_PREFIX=prusaslicer_
diff --git a/deps/Qhull/Qhull.cmake b/deps/Qhull/Qhull.cmake
index 497eb4b7c..e8d481834 100644
--- a/deps/Qhull/Qhull.cmake
+++ b/deps/Qhull/Qhull.cmake
@@ -2,7 +2,8 @@ include(GNUInstallDirs)
 bambustudio_add_cmake_project(Qhull
     URL "https://github.com/qhull/qhull/archive/v8.0.1.zip"
     URL_HASH SHA256=5287f5edd6a0372588f5d6640799086a4033d89d19711023ef8229dd9301d69b
-    CMAKE_ARGS 
+    CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5 
         -DINCLUDE_INSTALL_DIR=${CMAKE_INSTALL_INCLUDEDIR}
 )
 
diff --git a/deps/TBB/TBB.cmake b/deps/TBB/TBB.cmake
index 87b76828a..ed7c2a44b 100644
--- a/deps/TBB/TBB.cmake
+++ b/deps/TBB/TBB.cmake
@@ -3,7 +3,8 @@ bambustudio_add_cmake_project(
     URL "https://github.com/oneapi-src/oneTBB/archive/refs/tags/v2021.5.0.zip"
     URL_HASH SHA256=83ea786c964a384dd72534f9854b419716f412f9d43c0be88d41874763e7bb47
     #PATCH_COMMAND ${PATCH_CMD} ${CMAKE_CURRENT_LIST_DIR}/0001-TBB-GCC13.patch
-    CMAKE_ARGS          
+    CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5          
         -DTBB_BUILD_SHARED=OFF
         -DTBB_BUILD_TESTS=OFF
         -DTBB_TEST=OFF
diff --git a/deps/TIFF/TIFF.cmake b/deps/TIFF/TIFF.cmake
index 19bef192f..203a0def4 100644
--- a/deps/TIFF/TIFF.cmake
+++ b/deps/TIFF/TIFF.cmake
@@ -5,6 +5,7 @@ bambustudio_add_cmake_project(TIFF
     URL_HASH SHA256=6F3DBED9D2ECFED33C7192B5C01884078970657FA21B4AD28E3CDF3438EB2419
     DEPENDS ${ZLIB_PKG} ${PNG_PKG} ${JPEG_PKG}
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -Dlzma:BOOL=OFF
         -Dwebp:BOOL=OFF
         -Djbig:BOOL=OFF
diff --git a/deps/ZLIB/ZLIB.cmake b/deps/ZLIB/ZLIB.cmake
index c2dfd44af..49c6d6b86 100644
--- a/deps/ZLIB/ZLIB.cmake
+++ b/deps/ZLIB/ZLIB.cmake
@@ -9,6 +9,7 @@ bambustudio_add_cmake_project(ZLIB
   URL_HASH SHA256=c2856951bbf30e30861ace3765595d86ba13f2cf01279d901f6c62258c57f4ff
   PATCH_COMMAND ${patch_command}
   CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
     -DSKIP_INSTALL_FILES=ON         # Prevent installation of man pages et al.
     -DCMAKE_POSITION_INDEPENDENT_CODE=ON
 )
diff --git a/deps/libnoise/libnoise.cmake b/deps/libnoise/libnoise.cmake
index 246b2a918..449b94716 100644
--- a/deps/libnoise/libnoise.cmake
+++ b/deps/libnoise/libnoise.cmake
@@ -1,7 +1,8 @@
 # libnoise - coherent noise generation library for structured fuzzy skin
 # Source: https://github.com/bambulab/libnoise 
-
 bambustudio_add_cmake_project(libnoise
     URL https://github.com/bambulab/libnoise/archive/refs/tags/v1.0.0.zip
     URL_HASH SHA256=7aae3433e4c841a1b15ddf40dc995b3d32efb154b92b6796888002e275bff859
+    CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
 )
diff --git a/deps/wxWidgets/wxWidgets.cmake b/deps/wxWidgets/wxWidgets.cmake
index c47db9b6b..9d44d9498 100644
--- a/deps/wxWidgets/wxWidgets.cmake
+++ b/deps/wxWidgets/wxWidgets.cmake
@@ -27,7 +27,9 @@ bambustudio_add_cmake_project(wxWidgets
     GIT_REPOSITORY "https://github.com/bambulab/wxWidgets"
     GIT_TAG master
     DEPENDS ${PNG_PKG} ${ZLIB_PKG} ${EXPAT_PKG} ${TIFF_PKG} ${JPEG_PKG}
+    PATCH_COMMAND /usr/local/src/BambuStudio/deps/wxWidgets/fix-cmake.sh <SOURCE_DIR>
     CMAKE_ARGS
+        -DCMAKE_POLICY_VERSION_MINIMUM=3.5
         -DwxBUILD_PRECOMP=ON
         ${_wx_toolkit}
         "-DCMAKE_DEBUG_POSTFIX:STRING="
diff --git a/sandboxes/opencsg/CMakeLists.txt b/sandboxes/opencsg/CMakeLists.txt
index ace8f4d53..fd1e5601e 100644
--- a/sandboxes/opencsg/CMakeLists.txt
+++ b/sandboxes/opencsg/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 
 project(OpenCSG-example)
 
diff --git a/src/Shiny/CMakeLists.txt b/src/Shiny/CMakeLists.txt
index abdb96a72..a4e3bbea5 100644
--- a/src/Shiny/CMakeLists.txt
+++ b/src/Shiny/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(Shiny)
 
 add_library(Shiny STATIC
diff --git a/src/admesh/CMakeLists.txt b/src/admesh/CMakeLists.txt
index 217976318..645495d9b 100644
--- a/src/admesh/CMakeLists.txt
+++ b/src/admesh/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(admesh)
 
 add_library(admesh STATIC
diff --git a/src/boost/CMakeLists.txt b/src/boost/CMakeLists.txt
index e8c9e11ce..c8950e150 100644
--- a/src/boost/CMakeLists.txt
+++ b/src/boost/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(nowide)
 
 add_library(nowide STATIC
diff --git a/src/clipper/CMakeLists.txt b/src/clipper/CMakeLists.txt
index f62508820..5f8675e93 100644
--- a/src/clipper/CMakeLists.txt
+++ b/src/clipper/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(clipper)
 
 add_library(clipper STATIC
diff --git a/src/earcut/CMakeLists.txt b/src/earcut/CMakeLists.txt
index 08c7bb1be..13ad5c8f4 100644
--- a/src/earcut/CMakeLists.txt
+++ b/src/earcut/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.5)
 project(earcut_hpp LANGUAGES CXX C)
 
 option(EARCUT_BUILD_TESTS "Build the earcut test program" ON)
diff --git a/src/glu-libtess/CMakeLists.txt b/src/glu-libtess/CMakeLists.txt
index 8fca992a3..bfa43d369 100644
--- a/src/glu-libtess/CMakeLists.txt
+++ b/src/glu-libtess/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(glu-libtess)
 
 add_library(glu-libtess STATIC 
diff --git a/src/imgui/CMakeLists.txt b/src/imgui/CMakeLists.txt
index 592e78c8e..a358a83fa 100644
--- a/src/imgui/CMakeLists.txt
+++ b/src/imgui/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(imgui)
 
 add_library(imgui STATIC
diff --git a/src/imguizmo/CMakeLists.txt b/src/imguizmo/CMakeLists.txt
index fec443180..9d5c8edd6 100644
--- a/src/imguizmo/CMakeLists.txt
+++ b/src/imguizmo/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(imguizmo)
 
 add_library(imguizmo STATIC
diff --git a/src/libigl/CMakeLists.txt b/src/libigl/CMakeLists.txt
index f023826a5..f180ea416 100644
--- a/src/libigl/CMakeLists.txt
+++ b/src/libigl/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 project(libigl)
 
 add_library(libigl INTERFACE)
diff --git a/src/minilzo/CMakeLists.txt b/src/minilzo/CMakeLists.txt
index c5122ccf0..5ab2128c3 100644
--- a/src/minilzo/CMakeLists.txt
+++ b/src/minilzo/CMakeLists.txt
@@ -1,5 +1,5 @@
 project(minilzo)
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 
 add_library(minilzo INTERFACE)
 
diff --git a/src/miniz/CMakeLists.txt b/src/miniz/CMakeLists.txt
index 04d562b76..3beefa392 100644
--- a/src/miniz/CMakeLists.txt
+++ b/src/miniz/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(miniz)
 
 add_library(miniz INTERFACE)
diff --git a/src/qhull/CMakeLists.txt b/src/qhull/CMakeLists.txt
index ab9aba9af..310976eb2 100644
--- a/src/qhull/CMakeLists.txt
+++ b/src/qhull/CMakeLists.txt
@@ -28,7 +28,7 @@ endif()
 else(Qhull_FOUND)
 
 project(qhull)
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.5)
 
 # Define qhull_VERSION in CMakeLists.txt, Makefile, qhull-exports.def, qhull_p-exports.def, qhull_r-exports.def, qhull-warn.pri
 set(qhull_VERSION2 "2015.2 2016/01/18")  # not used, See global.c, global_r.c, rbox.c, rbox_r.c
diff --git a/src/semver/CMakeLists.txt b/src/semver/CMakeLists.txt
index 4b61a7456..59f12a4ca 100644
--- a/src/semver/CMakeLists.txt
+++ b/src/semver/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
 project(semver)
 
 add_library(semver STATIC
diff --git a/tests/cpp17/CMakeLists.txt b/tests/cpp17/CMakeLists.txt
index 4e151ecbf..d535f2083 100644
--- a/tests/cpp17/CMakeLists.txt
+++ b/tests/cpp17/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 project(Cpp17Test)
 
